### ISSUE TITLE
Namespace collisions

### ISSUE THREAD

**Issue by crockeea** [jcristovao/newtype-generics#1]
While this package is great and supercedes the 'newtype' package, 'newtype' was first to the party and has already claimed the `Control.Newtype` module name.  I have `newtype` installed as a dependency for `syntactic-2.0`, `type-natural-0.2.3.1`, and `constraints-0.4`, so when I import `Control.Newtype`, GHC complains that the module name is ambiguous.

The only solution I'm aware of would be using [package qualified imports](http://www.haskell.org/ghc/docs/latest/html/users_guide/syntax-extns.html#package-imports)

I propose moving this package to `Control.Newtype.Generic` or something similar, like the approach taken [here](https://hackage.haskell.org/package/deepseq-generics).

**Comment by sjakobi** [comment]
@crockeea Do you still run into these module name collisions these days? The packages that you listed seem to have removed their dependencies on `newtype`.

From a cursory look it also seems to me that only a few of `newtype`'s reverse dependencies have recent uploads, giving me hope that fewer and fewer people will have `newtype` in their dependency graphs.

**Comment by crockeea** [comment]
No,  I'm no longer using this package since the issue wasn't resolved.

**Comment by sjakobi** [comment]
I've finally changed my mind on this issue, moved the code to `Control.Newtype.Generics`, and deprecated `Control.Newtype`.

See the [new release](https://hackage.haskell.org/package/newtype-generics-0.5.3).
